tiberius/src/lib.rs

```diff
@@ -640,44 +641,68 @@ impl<'a> ToConnectEndpoint<Box<BoxableIo>, DynamicConnectionTarget> for &'a str
         let mut connect_params = ConnectParams::new();
 
         while !input.is_empty() {
-            let key_end = input.bytes().position(|x| x == b'=');
-            let (key, key_end) = match key_end {
-                None => return Err(TdsError::Conversion("connection string expected key. expected `=` never found".into())),
-                Some(key_end) => ((&input[..key_end]).to_lowercase(), key_end)
+            // (MSDN) The basic format of a connection string includes a series of keyword/value
+            // pairs separated by semicolons. The equal sign (=) connects each keyword and its value.
+            let key = {
+                let mut t = input.splitn(2, '=');
+                let key = t.next().unwrap();
```

Comment:
Don't panic there, return an error.

Reply:
No. If it panics here, it is a bug in this code and not a wrong input. The `splitn()` iterator always returns a valid first value (maybe an empty string), so the first `next()` never returns `None` regardless of the value of `input`.

Reply:
Oh right for this one, not sure if this shows a helpful message for malformed keys, but should be fine.